SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSubviews.cpp

```diff
@@ -0,0 +1,216 @@
+{
+	static bool IsAutonomousActorEntity(const Worker_EntityId EntityId, const EntityViewElement& Element);
+
+	static TArray<FDispatcherRefreshCallback> GetCallbacks(ViewCoordinator& Coordinator);
+};
+
+struct FSimulatedSubviewSetup
+{
+	static bool IsSimulatedActorEntity(const Worker_EntityId EntityId, const EntityViewElement& Entity);
+
+	static TArray<FDispatcherRefreshCallback> GetCallbacks(ViewCoordinator& Coordinator);
+};
+
+template <typename TCallable>
+FFilterPredicate GetRoleFilterPredicate(TCallable RolePredicate, USpatialNetDriver& NetDriver)
+{
+	return [RolePredicate, NetDriver = &NetDriver](const Worker_EntityId EntityId, const EntityViewElement& Entity) -> bool {
+		if (!FMainActorSubviewSetup::IsActorEntity(EntityId, Entity, *NetDriver))
+		{
+			return false;
+		}
+
+		return Invoke(RolePredicate, EntityId, Entity);
+	};
+}
+
+static TArray<FDispatcherRefreshCallback> CombineCallbacks(TArray<FDispatcherRefreshCallback> Lhs,
+														   const TArray<FDispatcherRefreshCallback>& Rhs)
+{
+	Lhs.Append(Rhs);
+	return Rhs;
+}
+
+bool FAuthoritySubviewSetup::IsAuthorityActorEntity(const Worker_EntityId EntityId, const EntityViewElement& Element)
+{
+	return Element.Authority.Contains(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID);
+}
+
+TArray<FDispatcherRefreshCallback> FAuthoritySubviewSetup::GetCallbacks(ViewCoordinator& Coordinator)
+{
+	return CombineCallbacks(FMainActorSubviewSetup::GetCallbacks(Coordinator),
+							{
+								Coordinator.CreateComponentExistenceRefreshCallback(SpatialConstants::PLAYER_CONTROLLER_COMPONENT_ID),
+								Coordinator.CreateComponentExistenceRefreshCallback(Partition::ComponentId),
+								Coordinator.CreateAuthorityChangeRefreshCallback(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID),
+							});
+}
+
+bool FAutonomousSubviewSetup::IsAutonomousActorEntity(const Worker_EntityId EntityId, const EntityViewElement& Element)
+{
+	return !FAuthoritySubviewSetup::IsAuthorityActorEntity(EntityId, Element)
+		   && Element.Authority.Contains(SpatialConstants::CLIENT_AUTH_COMPONENT_SET_ID);
+}
+
+TArray<FDispatcherRefreshCallback> FAutonomousSubviewSetup::GetCallbacks(ViewCoordinator& Coordinator)
+{
+	return CombineCallbacks(FAuthoritySubviewSetup::GetCallbacks(Coordinator),
+							{
+								Coordinator.CreateAuthorityChangeRefreshCallback(SpatialConstants::CLIENT_AUTH_COMPONENT_SET_ID),
+							});
+}
+
+bool FSimulatedSubviewSetup::IsSimulatedActorEntity(const Worker_EntityId EntityId, const EntityViewElement& Entity)
+{
+	return !FAuthoritySubviewSetup::IsAuthorityActorEntity(EntityId, Entity)
+		   && !FAutonomousSubviewSetup::IsAutonomousActorEntity(EntityId, Entity);
+}
+
+TArray<FDispatcherRefreshCallback> FSimulatedSubviewSetup::GetCallbacks(ViewCoordinator& Coordinator)
+{
+	return FAutonomousSubviewSetup::GetCallbacks(Coordinator);
+}
+
+FSubView& CreateActorSubView(USpatialNetDriver& NetDriver)
+{
+	return CreateCustomActorSubView({}, {}, {}, NetDriver);
```

Comment:
I don't know if this is a thing in GDK code, but in cases like these I find `/* argument name */` comments extremely useful.

Reply:
Yeah I agree - in this case however, the other method is just below, so I didn't think to write arg names here. Will add.